yb-voyager/cmd/live_migration.go

```diff
@@ -88,10 +116,60 @@ func handleEvent(event *tgtdb.Event) error {
 	if err != nil {
 		return fmt.Errorf("error transforming event key fields: %v", err)
 	}
-	batch := []*tgtdb.Event{event}
-	err = tdb.ExecuteBatch(batch)
-	if err != nil {
-		return fmt.Errorf("error executing batch: %v", err)
-	}
+	insertIntoTargetEventChan(event, targetEventChans)
 	return nil
 }
+
+func insertIntoTargetEventChan(e *tgtdb.Event, targetEventChans []chan *tgtdb.Event) {
+	// hash event
+	delimiter := "-"
+	stringToBeHashed := e.SchemaName + delimiter + e.TableName
+	for _, value := range e.Key {
+		stringToBeHashed += delimiter + *value
+	}
+	hash := fnv.New64a()
+	hash.Write([]byte(stringToBeHashed))
+	hashValue := hash.Sum64()
```

Comment:
Too many memory allocations!! Instead of building one giant string to be hashed, can we write each individual string into the `hash` object?
No need of delimiter.
Ordering of keys inside the `e.Key` matters.
I assume the hash function is not too compute intensive.

Reply:
didn't do too much research into hash functions, but the fnv was designed to be fast to compute. Didn't stand out in the profile either. 

https://en.wikipedia.org/wiki/Fowler%E2%80%93Noll%E2%80%93Vo_hash_function#Non-cryptographic_hash